### PR TITLE
SHIELD-37 SHIELD-42 Documentation for automate 4.x upgrade

### PR DESCRIPTION
**Pull request by vivekshankar1**
Signed-off-by: Vivek Shankar <vshankar@progress.com>

### :white_check_mark: Checklist

**All PRs** must tick these:

- [ ] I have read the [CONTRIBUTING document](https://github.com/chef/automate/blob/master/CONTRIBUTING.md).
- [ ] All commits signed-off for the [Developer Certification of Origin](https://github.com/chef/chef/blob/master/CONTRIBUTING.md#developer-certification-of-origin-dco).

With occasional exceptions, all PRs **from Progress employees** must tick these:

- [ ] Is the code clear? *(complicated code or lots of comments--subdivide and use well-named methods, meaningful variable names, etc.)*
- [ ] Consistency checked? *(user notifications, user prompts, visual patterns, code patterns, variable names)*
- [ ] Repeated code blocks eliminated? *(adapt and reuse existing components, blocks, functions, etc.)*
- [ ] Spelling, grammar, typos checked? *(at a minimum use `make spell` in any component directory)*
- [ ] Code well-formatted? *(indents, line breaks, etc. improve rather than hinder readability)*

All PRs **from Progress employees** should tick these if appropriate:

- [ ] Tests added/updated? (all new code needs new tests)
- [ ] Docs added/updated? (all customer-facing changes)

*Please add a note next to any checkbox above if you are NOT ticking it.*

### CODE REVIEW

File: components/docs-chef-io/content/automate/major_upgrade_4.x.md [hunk 1]
```diff
@@ -0,0 +1,315 @@
...
+Chef Automate provides an entire suite of enterprise capabilities for node visibility and compliance. Chef Automate upgrades from one minor version to another automatically. However, Chef Automatate will not automatically upgrade to a major version. See the instructions below for manually upgrading Chef Automate from date-based versions to Chef Automate 4.x.
+
+## Upgrade Journey
+
+Please choose following upgrade journey based on your current version of Chef Automate. All of these upgrades are manual upgrades.
+
+| Your Current Version | Upgrade To |
+| -------------------- | ---------- |
+| Any version before 20220329091442| 20220329091442|
+| 20220329091442| 3.0.x|
```
Comment (Dmaddu) [inline components/docs-chef-io/content/automate/major_upgrade_4.x.md:24]: I believe it's not 3.0.x, it should be a milestone version of 3.x.x

File: components/docs-chef-io/content/automate/major_upgrade_4.x.md [hunk 2]
```diff
@@ -0,0 +1,315 @@
...
+1. Manual upgrade to 3.0.x.
+1. Manual upgrade to 4.0.x
+
+## Prerequisites
+
+- **Plan your downtime**: This upgrade requires downtime. Before upgrading, set the environment to handle the downtime. Run `sudo chef-automate maintenance on` to turn on maintenance mode.
+- **Backup Chef Automate database**: This Chef Automate version upgrades ElasticSearch. [Backup](/automate/backup/) your data before upgrading.
+- **Current Version should be 3.0.x** If you are not on this version, please do normal upgrade as per your topology.
+
+## Upgrade to version 3.0.x
```
Comment (Dmaddu) [inline components/docs-chef-io/content/automate/major_upgrade_4.x.md:39]: insted from line num:39 to 79, can't we provide the link to 3.x.x upgrade.

File: components/docs-chef-io/content/automate/major_upgrade_4.x.md [hunk 3]
```diff
@@ -0,0 +1,315 @@
...
+
+Sharding needs to be disabled for automate running embedded elasticsearch. Please accept the checklist item asking permission to disable sharding.
+
+{{< /warning >}}
+
+### Chef Automate With Embedded Elasticsearch
+
+To upgrade Chef Automate with embedded Elasticsearch, follow the steps given below:
+
+**Upgrade Chef Automate from version 3.0.x to 4.0.x**
```
Comment (Dmaddu) [inline components/docs-chef-io/content/automate/major_upgrade_4.x.md:113]: 3.X.X

File: components/docs-chef-io/content/automate/major_upgrade_4.x.md [hunk 4]
```diff
@@ -0,0 +1,315 @@
...
+
+1. Start a major version upgrade:\
```
Comment (Dmaddu) [inline components/docs-chef-io/content/automate/major_upgrade_4.x.md:115]: how about mentioning the pre checklists as mandatory steps to do before the upgrade in the doc, so they will perform before their upgrade starts.

File: components/docs-chef-io/content/automate/major_upgrade_4.x.md (hunk 1 above)
Comment (vivekshankar1) [reply components/docs-chef-io/content/automate/major_upgrade_4.x.md]: done

File: components/docs-chef-io/content/automate/major_upgrade_4.x.md (hunk 2 above)
Comment (vivekshankar1) [reply components/docs-chef-io/content/automate/major_upgrade_4.x.md]: done

File: components/docs-chef-io/content/automate/major_upgrade_4.x.md (hunk 3 above)
Comment (vivekshankar1) [reply components/docs-chef-io/content/automate/major_upgrade_4.x.md]: done

File: components/docs-chef-io/content/automate/major_upgrade_4.x.md (hunk 4 above)
Comment (vivekshankar1) [reply components/docs-chef-io/content/automate/major_upgrade_4.x.md:81]: its given under prerequisites

File: components/docs-chef-io/content/automate/architectural_overview_os.md [hunk 5]
```diff
@@ -0,0 +1,79 @@
++++
```
Comment (karunesh-tech) [inline components/docs-chef-io/content/automate/architectural_overview_os.md:1]: We should update the existing architecture page with OpenSearch updates. No new page for architecture.

File: components/docs-chef-io/content/automate/install_os.md [hunk 6]
```diff
@@ -0,0 +1,380 @@
++++
```
Comment (karunesh-tech) [inline components/docs-chef-io/content/automate/install_os.md:1]: Replace the existing content of Elasticsearch with OpenSearch on the same exiting page. Drop this new page.

File: components/docs-chef-io/content/automate/architectural_overview.md [hunk 7]
```diff
@@ -14,7 +14,7 @@ gh_repo = "automate"
 
 ## Automate 2 Architecture
 
-![Automate 2 Architecture](/images/automate/a2-architecture.png)
+![Automate 2 Architecture](/images/automate/a2-architecture-os.JPEG)
```
Comment (dishanktiwari2501) [inline components/docs-chef-io/content/automate/architectural_overview.md:17]: We have to replace the JPEG with png file.

File: components/docs-chef-io/content/automate/opensearch_external_upgrade.md [hunk 8]
```diff
@@ -0,0 +1,112 @@
++++
+title = "Upgrade to External Opensearch"
+date = 2022-01-04T12:09:09-08:00
+draft = false
+
+[menu]
+  [menu.automate]
+    title = "Upgrade to Opensearch"
+    identifier = "automate/configuring_automate/opensearch/upgrade_to_opensearch.md opensearch_external_upgrade.md"
```
Comment (vivek-yadav) [inline components/docs-chef-io/content/automate/opensearch_external_upgrade.md:9]: `"` is missing at the end

**Review by jayvikramsharma1 — APPROVED** [review]

**Review by jayvikramsharma1 — APPROVED** [review]

**Review by punitmundra — APPROVED** [review]

File: components/docs-chef-io/content/automate/backup.md [hunk 9]
```diff
@@ -62,7 +62,11 @@ The [configured backup directory]({{< ref "backup.md#backup-to-a-filesystem" >}}
 
 A timestamp-based directory has a date-based name, such as `20180518010336`, in the `automate-elasticsearch-data` directory.
 
-To provide externally-deployed Elasticsearch nodes access to Chef Automate's built-in backup storage services, you must [configure Elasticsearch backup]({{< relref "install.md#configuring-external-elasticsearch" >}}) settings separately from Chef Automate's primary backup settings.
+<<<<<<< HEAD
+To provide externally-deployed Opensearch nodes access to Chef Automate's built-in backup storage services, you must [configure Opensearch backup]({{< relref "install_os.md#configuring-external-opensearch" >}}) settings separately from Chef Automate's primary backup settings.
+=======
+To provide externally-deployed Opensearch nodes access to Chef Automate's built-in backup storage services, you must [configure Opensearch backup]({{< relref "install.md#configuring-external-opensearch" >}}) settings separately from Chef Automate's primary backup settings.
+>>>>>>> 7a608c2859437526e91abfd34e3f9ab2678817ee
```
Comment (IanMadd) [inline components/docs-chef-io/content/automate/backup.md:69]: I'm not sure which of these is correct, but there's no install_os.md page, so the build failed.

**Pull request merged**